test/dynamics/models/test_rotating_frame.py

```diff
@@ -537,45 +540,46 @@ def test_state_transformations_no_frame_Operator_types(self):
         y = Operator([1.0, 1j])
         out = rotating_frame.state_into_frame(t, y)
         self.assertAllClose(out, y)
-        self.assertTrue(isinstance(out, np.ndarray))
+        self.assertArrayType(out)
         out = rotating_frame.state_out_of_frame(t, y)
         self.assertAllClose(out, y)
-        self.assertTrue(isinstance(out, np.ndarray))
+        self.assertArrayType(out)
 
         t = 100.12498
         y = Operator(np.eye(2))
         out = rotating_frame.state_into_frame(t, y)
         self.assertAllClose(out, y)
-        self.assertTrue(isinstance(out, np.ndarray))
+        self.assertArrayType(out)
         out = rotating_frame.state_out_of_frame(t, y)
         self.assertAllClose(out, y)
-        self.assertTrue(isinstance(out, np.ndarray))
+        self.assertArrayType(out)
 
     def test_state_transformations_no_frame_array_type(self):
         """Test frame transformations with no frame."""
 
         rotating_frame = RotatingFrame(None)
 
         t = 0.123
-        y = Array([1.0, 1j])
+        y = np.asarray([1.0, 1j])
         out = rotating_frame.state_into_frame(t, y)
         self.assertAllClose(out, y)
-        self.assertTrue(isinstance(out, Array))
+        self.assertEqual(DYNAMICS_NUMPY_ALIAS.infer_libs(out)[0], self.array_library())
         out = rotating_frame.state_out_of_frame(t, y)
         self.assertAllClose(out, y)
-        self.assertTrue(isinstance(out, Array))
+        self.assertEqual(DYNAMICS_NUMPY_ALIAS.infer_libs(out)[0], self.array_library())
 
         t = 100.12498
-        y = Array(np.eye(2))
+        y = self.asarray(np.eye(2))
         out = rotating_frame.state_into_frame(t, y)
         self.assertAllClose(out, y)
-        self.assertTrue(isinstance(out, Array))
+        self.assertEqual(DYNAMICS_NUMPY_ALIAS.infer_libs(out)[0], self.array_library())
         out = rotating_frame.state_out_of_frame(t, y)
         self.assertAllClose(out, y)
-        self.assertTrue(isinstance(out, Array))
+        self.assertEqual(DYNAMICS_NUMPY_ALIAS.infer_libs(out)[0], self.array_library())
 
 
-class TestRotatingJAXBCOO(QiskitDynamicsTestCase, TestJaxBase):
+@partial(test_array_backends, array_libraries=["jax_sparse"])
+class TestRotatingFrameJAXBCOO:
```

Comment:
It would be nice to change this into a class that tests both `"jax_sparse"` and `"scipy_sparse"`. This is a bit subtle in terms of how these functions are used in other parts of Dynamics, so I will make this change myself and open a PR to this one.